src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T):GroupsMap<T> {
+  return items.reduce((acc: GroupsMap<T>, el) => {
+    const keyValue = el[key] as string;
```

Comment:
Avoid using `as` in TS
```suggestion
    const keyValue = String(el[key]);
```